jetty-security/src/main/java/org/eclipse/jetty/security/DefaultAuthenticatorFactory.java

```diff
@@ -62,7 +62,7 @@ public Authenticator getAuthenticator(Server server, ServletContext context, Aut
         String auth = configuration.getAuthMethod();
         Authenticator authenticator = null;
 
-        if (auth == null || Constraint.__BASIC_AUTH.equalsIgnoreCase(auth))
+        if (Constraint.__BASIC_AUTH.equalsIgnoreCase(auth))
```

Comment:
Is there a test for this case?

Reply:
@lachlan-roberts I agree that we need at least some simple test that checks a DIS is created if there is no realm.  Ie not necessarily testing this specific line, but testing the PR in general.